src/Cli/LegacyParamArgvInput.php

```diff
@@ -0,0 +1,168 @@
+        }
+        return $this->hasLegacyParameterOption($values);
+    }
+
+    public function getParameterOption(string|array $values, string|bool|int|float|array|null $default = false, bool $onlyParams = false): mixed
+    {
+        if (parent::hasParameterOption($values, $onlyParams)) {
+            return parent::getParameterOption($values, $default, $onlyParams);
+        }
+        return $this->getLegacyParameterOption($values, $default);
+    }
+
+    /**
+     * Binds the current Input instance with the given arguments and options.
+     *
+     * Also converts any arg-style params into true flags, based on the options defined.
+     */
+    public function bind(InputDefinition $definition): void
+    {
+        // Convert arg-style params into flags
+        $tokens = $this->argv;
+        $convertedFlags = [];
+        $hadLegacyParams = false;
+        foreach ($definition->getOptions() as $option) {
+            $flagName = '--' . $option->getName();
+            // Check if there is a legacy param first. This saves us from accidentally getting
+            // values that come after the end of options (--) signal
+            if (!$this->hasLegacyParameterOption($flagName)) {
+                continue;
+            }
+            // Get the value from the legacy param
+            $value = $this->getLegacyParameterOption($flagName);
+            if ($value && !$this->hasLegacyParameterOption($flagName . '=' . $value)) {
+                // symfony/console will try to get the value from the next argument if the current argument ends with `=`
+                // We don't want to count that as the value, so double check it.
+                $value = null;
+            } elseif ($option->acceptValue()) {
+                if ($value === '' || $value === null) {
+                    $convertedFlags[] = $flagName;
+                } else {
+                    $convertedFlags[] = $flagName . '=' . $value;
+                }
+            } else {
+                // If the option doesn't accept a value, only add the flag if the value is true.
+                $valueAsBool = filter_var($value, FILTER_VALIDATE_BOOL, FILTER_NULL_ON_FAILURE) ?? true;
+                if ($valueAsBool) {
+                    $convertedFlags[] = $flagName;
+                }
+            }
+            $hadLegacyParams = true;
+            // Remove the legacy param from the token set
+            foreach ($tokens as $i => $token) {
+                if (str_starts_with($token, $option->getName() . '=')) {
+                    unset($tokens[$i]);
+                    break;
+                }
+            }
+        }
+        if (!empty($convertedFlags)) {
+            // Make sure it's before the end of options (--) signal if there is one.
+            $tokens = ArrayLib::insertBefore($tokens, $convertedFlags, '--', true, true);
+        }
+        if ($hadLegacyParams) {
+            // We only want the warning once regardless of how many params there are.
```

Comment:
It's a warning.

Reply:
It's an `E_USER_DEPRECATED`? Which is neither. Since we say 'notice' everywhere we should just refer to it as a notice rather than a warning.

Reply:
It doesn't really matter, it's a comment and we all know what it means. I'm gonna change it at this stage just because arguing about this is a waste of time.